### ISSUE TITLE
Allow for filtering out specific measurement entities through regular expressions

### ISSUE THREAD

**Issue by Sebastiaan127001** [ICTU/quality-time#3328]
This feature already exists for OWASP sources, I believe, but it would also be useful for other sources like AXE core.

For example, AXE core has limitations and therefore generates false positives. 
Some tools cannot be configured properly to remove these violations. In those cases, it would be useful to manage this in QT. That is the added value of QT.

Add two fields:
- Entities to include (regex)
- Entities to exclude (regex)

Add a possibility to the data model to exclude entity attributes from matching.

Implement a generic matcher in the base collector.

Note: Order of including/excluding doesn't matter, explain this in the UI somewhere.

We first need to refactor the collector to allow for filtering entities by regular expression in one location (in the base SourceCollector): #4714 and #5240.

This is related to https://github.com/ICTU/quality-time/issues/4615

**Comment by Sebastiaan127001** [comment]
this would also come in handy when reviewing individual SonarQube violations. Sometimes the team has decided that they won't fix violations of a certain kind or they don't see it as an issue.

**Comment by fniessink** [comment]
Blocked by #3332 because we have to change the same code for both feature requests.

**Comment by Sebastiaan127001** [comment]
for now, yes

**Comment by Sebastiaan127001** [comment]
![image](https://user-images.githubusercontent.com/95095453/169051786-f17ce0ba-26ac-4008-8eee-5aaae3c6bb06.png)

The above example shows that the issues reappearing.

**Comment by Sebastiaan127001** [comment]
%VOTE

**Comment by Sebastiaan127001** [comment]
@fniessink can we combine this with the issues that we have with software versions that do not comply to the standard?
Currently, the software version metric reports an error if the version has a postfix.
If we can add the regular expression to the source configuration, we might be able to fix this in one go.

**Comment by wkoot** [comment]
@Sebastiaan127001 can you make separate issues for changes to the SonarQube and software_versions metrics, if this is still applicable?
